### PR TITLE
Save and restore map state on rotation for way name and camera tracking

### LINKED ISSUE #1055: When map scrolled wayname not hidden after screen rotation
Noticed that if you swipe the map the wayname underneath the puck disappears (that's ok) but if you then rotate the screen it appears when it should be still hidden 👀 

wayname_does_not_survive_rotation

### PR DESCRIPTION
**Pull request by danesfeder**
Closes #1055 and addresses part of https://github.com/mapbox/mapbox-navigation-android/issues/1072

TODO:
- [x] Javadoc for `NavigationMapboxMap`
- [x] Tests

### CODE REVIEW

**Comment by danesfeder** [comment]
@Guardiola31337 this is ready for review - I added javadoc for `NavigationMapboxMap` and some instrumentation tests for rotation.

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 1]
```diff
@@ -75,85 +94,248 @@ public NavigationMapboxMap(MapView mapView, MapboxMap mapboxMap) {
...
+  /**
+   * Can be used to store the current state of the map in
+   * {@link android.support.v4.app.FragmentActivity#onSaveInstanceState(Bundle, PersistableBundle)}.
+   * <p>
+   * This method uses {@link NavigationMapboxMapInstanceState}, stored with the
+   * {@link NavigationMapboxMap#MAP_INSTANCE_STATE_KEY}.
+   *
+   * @param outState to store state variables
+   */
+  public void onSaveInstanceState(Bundle outState) {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Really like how we took advantage of the naming in `restoreFrom` to make the API easier to read and understand ❤️
Have we considered doing something similar here?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 2]
```diff
@@ -144,12 +149,12 @@ public void onSaveInstanceState(Bundle outState) {
...
   public void onRestoreInstanceState(Bundle savedInstanceState) {
-    NavigationViewInstanceState navigationViewInstanceState = savedInstanceState
-      .getParcelable(getContext().getString(R.string.instance_state));
-
+    String instanceKey = getContext().getString(R.string.navigation_view_instance_state);
+    NavigationViewInstanceState navigationViewInstanceState = savedInstanceState.getParcelable(instanceKey);
     recenterBtn.setVisibility(navigationViewInstanceState.getRecenterButtonVisibility());
     resetBottomSheetState(navigationViewInstanceState.getBottomSheetBehaviorState());
     updateInstructionListState(navigationViewInstanceState.isInstructionViewVisible());
+    mapInstanceState = savedInstanceState.getParcelable(MAP_INSTANCE_STATE_KEY);
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Have we considered adding this into `NavigationMapboxMap` or passing the key to `NavigationMapboxMap#onSaveInstanceState` somehow?
Having `MAP_INSTANCE_STATE_KEY` as `public` in `NavigationMapboxMap` looks weird to me.

**Comment by danesfeder** [comment]
@Guardiola31337 this is ready for another round of 👀

**Pull request merged**